tests/test_router_quote.py

```diff
@@ -172,6 +184,46 @@ def test_transmuter_tokens(self, environment_url, amount):
 
         # Validate the quote test
         self.validate_quote_test(quote, amount, denom_in, spot_price_scaling_factor, expected_in_base_out_quote_price, expected_token_out, error_tolerance)
+    
+
+    @pytest.mark.parametrize("amount", [1000])
+    @pytest.mark.parametrize("token_pair", orderbook_token_pairs())
+    def test_orderbook(self, environment_url, amount, token_pair):
+        """
+        This test aims to validate the orderbook functionality by performing a direct quote request.
+        
+        The amount out from direct quote is compared to the amount out calculated by amount in after fee * spot price
+        to ensure that the orderbook quote is working as expected.
+
+        Small amount in is used since spot price is calculated based on the tick price at the top of the orderbook.
+        If the amount in is large, the actual amount out will be different from the amount out calculated by amount in after fee * spot price
+        due to the moving ticks and potential low liquidity on one side of the orderbook.
+        """
+        (pool_id, (denom_in, denom_out)) = token_pair
+
+        sqs_service = conftest.SERVICE_MAP[environment_url]
+        start_time = time.time()
+        response = sqs_service.get_custom_direct_quote(str(amount) + denom_in, denom_out, pool_id)
+        elapsed_time_ms = (time.time() - start_time) * 1000
+
+        assert response.status_code == 200, f"Error: {response.text}"
+        assert EXPECTED_LATENCY_UPPER_BOUND_MS > elapsed_time_ms, \
+             f"Error: latency {elapsed_time_ms} exceeded {EXPECTED_LATENCY_UPPER_BOUND_MS} ms, token in {denom_in} and token out {denom_out}" 
+
+        res = response.json()
+        
+        spot_price = float(res["in_base_out_quote_spot_price"])
```

Comment:
Can we use spot price from another data source (i.e. Numia)?

The reason is that we are computing the expected amount from the system under test.

If the system under test is broken, we might compute the wrong expected value and get a false positive